### PR TITLE
Run external tests with and without optimizer.

### PR DESCRIPTION
**Pull request by chriseth**
Closes https://github.com/ethereum/solidity/issues/6156

### CODE REVIEW

File: test/externalTests.sh [hunk 1]
```diff
@@ -91,10 +91,15 @@ function test_truffle
...
 EOF
 
-      npx truffle compile
-      echo "Verify that the correct version ($SOLCVERSION) of the compiler was used to compile the contracts..."
-      grep -e "$SOLCVERSION" -r build/contracts > /dev/null
-      npm run test
+      for optimize in "{enabled: false }" "{enabled: true }" "{enabled: true, details: { yul: true } }"
+      do
+        rm -rf build || true
+        echo "module.exports['compilers']['solc']['settings'] = {optimizer: $optimize };" >> truffle*.js
```
Comment (axic) [inline test/externalTests.sh:97]: Did you verify this passes through to jsonio?

Comment (chriseth) [reply test/externalTests.sh:97]: I hoped there would be some gas values, but I'll add a check to re-discover this in the generated metedata like we do with the compiler version above.

Comment (chriseth) [reply test/externalTests.sh:97]: Hm, it looks like this is not possible. I'll check manually on circle.

**Comment by chriseth** [comment]
I checked manually and found differences between run 1 and run 2 but not between 2 and 3. It might be because it does not use abiencoderv2.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**